### PR TITLE
initializeGCParameters During CRIU Restore

### PR DESCRIPTION
**Pull request by RSalman**
Relates to https://github.com/eclipse-openj9/openj9/pull/17459#discussion_r1231139601

Signed-off-by: Salman Rana <salman.rana@ibm.com>

### CODE REVIEW

**Comment by RSalman** [comment]
@amicic please review

File: gc/base/Configuration.cpp [hunk 1]
```diff
@@ -525,3 +525,27 @@ MM_Configuration::createParallelDispatcher(MM_EnvironmentBase *env, omrsig_handl
...
+
+	/* The initialization of the GC thread count can only happen if it is not enforced by the user. */
+	initializeGCThreadCount(env);
+
+	/* Currently, threads don't shutdown during reinitialization, so it is important to
+	 * prevent the thread count from dropping below the number of checkpoint threads.
+	 * This adjustment may no longer be necessary in the future once the shutdown behavior
+	 * of dispatcher threads during restore is thoroughly tested.
+	 */
+	extensions->gcThreadCount = OMR_MAX(
```
Comment (amicic) [inline gc/base/Configuration.cpp:543]: temporarily couple of those lines that are migrated from J9 will be executed twice, right? hopefully, not causing major problems

**Review by amicic — APPROVED** [review]

File: gc/base/Configuration.cpp (hunk 1 above)
Comment (RSalman) [reply gc/base/Configuration.cpp:543]: Yup, `initializeGCThreadCount` and MAX adjustment will be just done twice, no issue with that though.

**Comment by RSalman** [comment]
@babsingh this is ready for further with review / merge

**Comment by babsingh** [comment]
jenkins build all

File: gc/base/Configuration.hpp [hunk 2]
```diff
@@ -178,11 +178,7 @@ class MM_Configuration : public MM_BaseVirtual
 	 * @param[in] env the current environment.
 	 * @return boolean indicating whether the configuration was successfully updated.
 	 */
-	virtual bool reinitializeForRestore(MM_EnvironmentBase* env)
-	{
-		Assert_MM_unreachable();
-		return false;
-	}
+	virtual bool reinitializeForRestore(MM_EnvironmentBase *env);
```
Comment (babsingh) [inline gc/base/Configuration.hpp:181]: To be consistent with the existing format of the file.
```suggestion
	virtual bool reinitializeForRestore(MM_EnvironmentBase* env);
```

**Comment by babsingh** [comment]
jenkins build all

File: gc/base/Configuration.hpp (hunk 2 above)
Comment (amicic) [reply gc/base/Configuration.hpp]: the first style is actually more popular in overall OMR (and J9) code, so next time we touch this file I'd actually prefer to convert them all back

Comment (babsingh) [reply gc/base/Configuration.hpp]: Yes, `int32_t *x` is preferred over `int32_t* x`. @RSalman I still see a mixture of both styles in the PR. Can you make the changes consistent?

Comment (RSalman) [reply gc/base/Configuration.hpp]: I've converted all the touched files to use `int32_t* x` style its a lot more used in the files

**Comment by babsingh** [comment]
jenkins build all

**Comment by babsingh** [comment]
> https://ci.eclipse.org/omr/job/PullRequest-linux_riscv64_cross/1161/consoleFull

This failure is unrelated to this PR.

```
10:54:23  29/30 Test #29: comptest ..........................Child killed***Exception: 1803.66 sec
10:54:23  test 30
10:54:23        Start 30: compunittest
```

**Review by babsingh — APPROVED** [review]

**Pull request merged**